sdk/ml/azure-ai-ml/azure/ai/ml/_azure_environments.py

```diff
@@ -279,3 +276,13 @@ def _convert_arm_to_cli(arm_cloud_metadata):
             module_logger.warning("Property on cloud not found in arm cloud metadata: %s", ex)
             continue
     return cli_cloud_metadata_dict
+
+def _add_cloud_to_environments(kwargs):
+    _environments[kwargs["cloud"]] = {
```

Comment:
if this cloud is already set then we should not insert - this way we can't mess up the known-good hardcoded configs with potentially bad configs